### PR TITLE
Fix missing cast to hash in repo discovery script

### PR DESCRIPTION
**Pull request by howcanunot**
Fix this exception in `discovery-repos.rb` script
```
cam/steps/discover-repos.rb:125:in `<main>': undefined method `values' for an instance of Array (NoMethodError)
lines = [found.first[1].keys.join(',')] + found.values.map { |m| m.values.join(',') }
Did you mean?  values_at
```
https://github.com/yegor256/cam/issues/211

### CODE REVIEW

**Comment by yegor256** [comment]
@howcanunot thanks for the fix! Would be great to supplement the fix with a unit test. Otherwise, how do we know anything was fixed?

**Comment by howcanunot** [comment]
> @howcanunot thanks for the fix! Would be great to supplement the fix with a unit test. Otherwise, how do we know anything was fixed?

What is the best way to add such a test? Would it be enough to add my case to `tests/steps/test-discover-repos.sh`?

**Comment by howcanunot** [comment]
some changes in commit:

- in `dry_run` script mode we should get not only single repo but `page-size` for every iteration, just like in normal mode
- extra test to verify that we were able to grep a sufficiently large number of repos

**Comment by howcanunot** [comment]
@yegor256 Good afternoon! Can you please check this pr?

File: steps/discover-repos.rb [hunk 1]
```diff
@@ -75,10 +75,11 @@
     break
   end
   json = if opts[:dry]
-    { items: page > 100 ? [] : [{ full_name: "foo/#{Random.hex(5)}", created_at: Time.now }] }
+    { items: page > 100 ? [] : Array.new(size) { { full_name: "foo/#{Random.hex(5)}", created_at: Time.now } } }
   else
     github.search_repositories(query, per_page: size, page: page)
   end
+  puts(json)
```
Comment (yegor256) [inline steps/discover-repos.rb:82]: @howcanunot this is a debug output?

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**